# Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

### Задача 1
Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании.<br>
Найдите $M(X)$, $D(X)$.

#### Решение

Описана схема Бернулли, таким образом случайная величина $X$, число угаданных вопросов имеет биномиальное распределение с параметрами $p$ и $n$, где

* $n = 5$ — число независисмых случайных экспериментов, число вопросов в нашем случае,
* $p = \dfrac{1}{4}$ — вероятность угадать ответ на вопрос.

$$M(X) = n p = \dfrac{5}{4}$$

$$D(X) = npq = np(1-p) = 5\cdot \dfrac{1}{4}\cdot\dfrac{3}{4} = \frac{15}{16}$$

### Задача 2
Вероятность попадания в цель при одном выстреле равна $0.01$. Какова вероятность того, что число попаданий при $200$ выстрелах будет не менее $5$ и не более $10$?

#### Решение

Пусть $X$ - случайная величина количества попаданий в мишень при $200$ выстрелах.<br>
Она имеет биномиальное распределение с параметрами 
* $n = 200$, 
* $p = 0.01$.

Нам необходимо найти $P(5\leqslant X \leqslant 10)$:

$$P(5\leqslant X \leqslant 10) = \displaystyle\sum_{k=5}^{10} C_n^k\cdot p^k \cdot (1-p)^{n-k}$$

In [1]:
from scipy.special import comb

n = 200
p = 0.01
result = 0

sum(comb(n, k) * p**k * (1 - p)**(n - k) for k in range(5, 11))

0.051739381860374606

#### Ответ
Вероятность того, что число попаданий при $200$ выстрелах будет не менее $5$ и не более $10$ равно примерно $5\%$.

### Задача 3

В условиях предыдущей задачи сколько выстрелов нужно сделать, чтобы быть уверенным с вероятностью $0.9$, что хотя бы $10$ раз будет совершено попадание?

#### Решение

Нам необходимо найти минимальное $n$, при котором $P(X(n) \geqslant 10) \geqslant 0.9$.

Т.к. $P(X(n) \geqslant 10) = 1 - P(X(n) < 10)$, то нам достаточно найти минимальное $n$, при котором $P(X(n) < 10) < 0.1$.

$P(X(n) < 10) = \displaystyle\sum_{i=0}^{9}C_n^i\cdot p^i \cdot (1-p)^{n-i} = f(n)$ — функция от общего число выстрелов $n$, возвращающая вероятность того, что мы попали суммарно меньше $10$ раз.

Будем считать значение этой функции, увеличивая число $n$. Минимальное $n$, которое даст нам значение функции меньшее $0.1$ будет решением задачи.

In [2]:
k = 10  # - минимальное количество попаданий, необходимое нам.
p = 0.01
q = 1 - p

def f(n):
    return sum(comb(n, i) * p**i * q**(n - i) for i in range(10))

In [3]:
n = 10  # Начальное количество выстрелов. Меньше нет смысла брать, так как нам необходимо от 10 попаданий.

while f(n) >= 0.1:
    n += 1
    
n  # Как только n примет значение, при котором P(X < 10) < 0.1, цикл завершится. n будет равно в точности искомому числу.

1418

#### Ответ
Достаточно сделать $1418$ выстрелов, чтобы быть уверенным с вероятностью $90\%$, что хотя бы $10$ раз будет совершено попадание.